src/lib.rs

```diff
@@ -42,22 +60,63 @@ impl TsnSocket {
 
 const CONTROL_SOCK_PATH: &str = "/var/run/tsn.sock";
 
-fn send_cmd(command: String) -> Result<String, std::io::Error> {
-    let mut stream = UnixStream::connect(CONTROL_SOCK_PATH)?;
-    stream.write_all(command.as_bytes())?;
-    let mut msg = String::new();
-    stream.read_to_string(&mut msg)?;
-    Ok(msg)
+#[derive(Eq, PartialEq)]
+enum LockKind {
+    LOCK,
+    UNLOCK,
 }
 
-fn create_vlan(ifname: &str, vlanid: u16) -> Result<String, std::io::Error> {
-    let command = format!("create {} {}\n", ifname, vlanid);
-    send_cmd(command)
+fn create_vlan(ifname: &str, vlanid: u16) -> Result<i32, i32> {
+    let configs = config::read_config(CONFIG_PATH).unwrap();
+    let config = configs.get(ifname).unwrap();
+    let shm_name = format!("{}.{}", ifname, vlanid);
```

Comment:
shmem is not ours. use some prefix to avoid conflict from other apps like `libtsn_vlan_eth0.11`